hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/scheduler/fair/policies/FairSharePolicy.java

```diff
@@ -175,7 +179,13 @@ public boolean checkIfUsageOverFairShare(Resource usage, Resource fairShare) {
   }
 
   @Override
-  public byte getApplicableDepth() {
-    return SchedulingPolicy.DEPTH_ANY;
+  public boolean isChildPolicyAllowed(SchedulingPolicy childPolicy) {
+    if (childPolicy instanceof DominantResourceFairnessPolicy) {
+      LOG.info("Queue policies can't be " + DominantResourceFairnessPolicy.NAME
+          + " if the parent policy is " + getName() + ". Please choose "
+          + "other polices for child queues instead.");
```

Comment:
IMO, we should either (1) not say anything about other policies or (2) list the policies that are allowed.